### PR TITLE
Fix Infinite loop in the TM

### PR DESCRIPTION
**Pull request by MatanYadaev**
## **Type**
bug_fix, enhancement

___

## **Description**
- Added `alertType` field to the search schema to enhance filtering capabilities.
- Introduced `businessId` and `counterpartyId` fields for broader search criteria.
- Fixed re-rendering issue in Alert Analysis by adding a `key` prop.
- Removed a redundant navigation call to improve user experience.
- Corrected the mapping of alert states to statuses for accurate status representation.
- Standardized `AlertState` enum values in the database schema for consistency.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>get-alerts-search-schema.ts</strong><dd><code>Enhance Alerts Search Schema with New Fields</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/get-alerts-search-schema.ts
<li>Added <code>alertType</code> field to the search schema to support filtering by <br>alert type.<br> <li> Introduced optional <code>businessId</code> and <code>counterpartyId</code> fields.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2145/files#diff-5961638e1322569191613088e1fbd85783354f931fc50967d37b6c3b8dd0756e">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Bug_fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>TransactionMonitoringAlertsAnalysis.page.tsx</strong><dd><code>Fix Alert Analysis Sheet Re-rendering Issue</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/TransactionMonitoringAlertsAnalysis/TransactionMonitoringAlertsAnalysis.page.tsx
<li>Added <code>key</code> prop to <code>AlertAnalysisSheet</code> component to force re-render when <br><code>alertId</code> changes.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2145/files#diff-aeef0d6dfd8b1eeb8b0db7caccbfaadac9088a5f3fa9486471246abfe47f1cbc">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useTransactionMonitoringAlertsAnalysisPageLogic.tsx</strong><dd><code>Remove Redundant Navigation Call</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/TransactionMonitoringAlertsAnalysis/hooks/useTransactionMonitoringAlertsAnalysisPageLogic/useTransactionMonitoringAlertsAnalysisPageLogic.tsx
- Removed duplicate `navigateBack()` call in `useEffect` hook.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2145/files#diff-8f5ef1092896ccad900af14272b56702517c2230d0a9a0f6d1e6e3a70b959176">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>alert.service.ts</strong><dd><code>Correct Alert State to Status Mapping</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/alert/alert.service.ts
<li>Fixed mapping of <code>AlertState.under_review</code> to <code>AlertStatus.pending</code>.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2145/files#diff-c71af9f2ec363ee7d318b710abb76b35af4cdf715c63bdbc431212ef82cfcb5e">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>schema.prisma</strong><dd><code>Fix and Format AlertState Enum Values</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/prisma/schema.prisma
<li>Fixed enum values for <code>AlertState</code> to match the database mapping.<br> <li> Cleaned up formatting in the <code>AlertState</code> enum.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2145/files#diff-79f9532cd1d8e6c67881328fa550f851c8429f183de839b32575e959cdabef26">+6/-9</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by alonp99 — APPROVED** [review]

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlertsAnalysis/hooks/useTransactionMonitoringAlertsAnalysisPageLogic/useTransactionMonitoringAlertsAnalysisPageLogic.tsx [hunk 1]
```diff
@@ -30,12 +30,14 @@ export const useTransactionMonitoringAlertsAnalysisPageLogic = () => {
     }
 
     navigateBack();
+    navigateBack();
```
Comment (alonp99) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlertsAnalysis/hooks/useTransactionMonitoringAlertsAnalysisPageLogic/useTransactionMonitoringAlertsAnalysisPageLogic.tsx:33]: You should probably leave a comment here explaining the duplication

Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlertsAnalysis/hooks/useTransactionMonitoringAlertsAnalysisPageLogic/useTransactionMonitoringAlertsAnalysisPageLogic.tsx:33]: This change should be reverted.

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlertsAnalysis/TransactionMonitoringAlertsAnalysis.page.tsx [hunk 2]
```diff
@@ -6,11 +6,12 @@ import { valueOrNA } from '@/common/utils/value-or-na/value-or-na';
 import { ctw } from '@/common/utils/ctw/ctw';
 
 export const TransactionMonitoringAlertsAnalysisPage = () => {
-  const { transactions, onNavigateBack, alertDefinition, isLoadingAlertDefinition } =
+  const { transactions, onNavigateBack, alertDefinition, isLoadingAlertDefinition, alertId } =
     useTransactionMonitoringAlertsAnalysisPageLogic();
 
   return (
     <AlertAnalysisSheet
+      key={alertId}
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlertsAnalysis/TransactionMonitoringAlertsAnalysis.page.tsx:14]: This should be reverted with navigateBack

Comment (MatanYadaev) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlertsAnalysis/TransactionMonitoringAlertsAnalysis.page.tsx:14]: Will appreciate an explanation :)

**Pull request merged**